sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -387,8 +376,11 @@ tableProvider
 
 createTableClauses
     :((OPTIONS options=tablePropertyList) |
-     (PARTITIONED BY partitioning=transformList) |
+     (PARTITIONED BY partitioning=partitionFieldList) |
```

Comment:
I think an easier way to merge the two syntaxes is
```
(PARTITIONED BY partitioning=partitionFieldList) |
(PARTITIONED BY '(' partitionColumns=colTypeList ')' |
```
Then antlr can help us to guarantee that users can't mix different PARTITION BY syntaxes, and it requires fewer changes in `AstBuilder`.

Reply:
The result of doing this is a worse error message for a user because the parser would generate an error, instead of the user-friendly error message that is generated by `AstBuilder`. I think an overall worse experience for users isn't worth changing this.